x-pack/plugins/apm/common/environment_filter_values.ts

```diff
@@ -33,3 +33,18 @@ export const ENVIRONMENT_NOT_DEFINED = {
 export function getEnvironmentLabel(environment: string) {
   return environmentLabels[environment] || environment;
 }
+
+export function getEnvironmentUrlParam(
+  requestedEnvironment?: string,
+  queryEnvironment?: string
```

Comment:
```suggestion
  requestedEnvironment?: string,
  currentEnvironment?: string
```

Reply:
I wanted to keep this as `queryEnvironment` because how an empty value is interpreted is specific to whether it's a query parameter. Maybe `currentQueryEnvironment` is a good compromise here, but happy to go with your suggestion as well.